python/lsst/qserv/tests/benchmark.py

```diff
@@ -199,78 +229,94 @@ def _parseFile(self, qF, withQserv):
+    def loadData(self, mode, dbName):
+        """Loads data from input files located in caseXX/data/
+
+        Parameters
+        ----------
+        mode : str
+            One of "mysql" or "qserv"
+        dbName : str
+            Database name
         """
-        self.logger.info("Loading data from %s (%s mode)", self._in_dirname,
-                         self._mode)
+        dataLoader = self.connectAndInitDatabases(mode, dbName)
+        _LOG.info("Loading data from %s (%s mode)", self._in_dirname, mode)
         for table in self.dataReader.orderedTables:
-            self.dataLoader[self._mode].createLoadTable(table)
+            dataLoader.createLoadTable(table)
+        dataLoader.finalize()
 
     def cleanup(self):
-        """
-        Cleanup of previous tests temporary ant output files
+        """Cleanup of previous tests output files
         """
         if os.path.exists(self._out_dirname):
             shutil.rmtree(self._out_dirname)
         os.makedirs(self._out_dirname)
 
-    def connectAndInitDatabases(self):
-        self.logger.debug("Creation of data loader for %s mode", self._mode)
-        if (self._mode == 'mysql'):
-            self.dataLoader[self._mode] = mysqlDbLoader.MysqlLoader(
+    def connectAndInitDatabases(self, mode, dbName):
+        """Establish database server connection and create database.
+
+        Parameters
+        ----------
+        mode : str
+            One of "mysql" or "qserv"
+        dbName : str
+            Database name
+
+        Returns
+        -------
+        `DbLoader` instance to be used for data loading
+        """
+        _LOG.debug("Creation of data loader for %s mode", mode)
+        if mode == 'mysql':
+            dataLoader = mysqlDbLoader.MysqlLoader(
                 self.config,
                 self.dataReader,
-                self._dbName,
+                dbName,
                 self._multi_node,
                 self._out_dirname
             )
-        elif (self._mode == 'qserv'):
-            self.dataLoader[self._mode] = qservDbLoader.QservLoader(
+        elif mode == 'qserv':
+            dataLoader = qservDbLoader.QservLoader(
                 self.config,
                 self.dataReader,
-                self._dbName,
+                dbName,
                 self._multi_node,
                 self._out_dirname
             )
-        self.logger.debug("Initializing database for %s mode", self._mode)
-        self.dataLoader[self._mode].prepareDatabase()
-
-    def finalize(self):
-        if (self._mode == 'qserv'):
-            self.dataLoader['qserv'].workerInsertXrootdExportPath()
-
-            # xrootd is restarted by wmgr
-
-            # Reload Qserv (empty) chunk cache
-            self.dataLoader['qserv'].resetChunksCache()
-
-        # Close socket connections
-        del(self.dataLoader[self._mode])
+        _LOG.debug("Initializing database for %s mode", mode)
+        dataLoader.prepareDatabase()
```

Comment:
here, `dataLoader` will not exist if mode is not one of the 2 valid values, which could lead to misleading error messages.

Reply:
True, again it's not going to happen, but I'l add a test.